source/extensions/filters/listener/proxy_protocol/proxy_protocol.h

```diff
@@ -128,8 +132,7 @@ class Filter : public Network::ListenerFilter, Logger::Loggable<Logger::Id::filt
 
   ProxyProtocolVersion header_version_{Unknown};
 
-  // Stores the portion of the first line that has been read so far.
-  char buf_[MAX_PROXY_PROTO_LEN_V2];
+  Buffer::InstancePtr buffer_;
```

Comment:
Yeah fair enough, given all of the manual accounting and the extra heap allocation I guess I'm OK going back to what you had before. @ggreenway ?